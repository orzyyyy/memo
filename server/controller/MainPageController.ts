import fs from 'fs-extra';
import path from 'path';
import { Controller, Request } from '../utils/decorator';

@Controller('/')
export default class MainPageController {
  @Request({ url: '/', method: 'get' })
  async getMainPage(ctx: any) {
    ctx.type = 'html';
    ctx.response.body = fs.createReadStream(
      path.join(process.cwd(), 'dist', 'index.html'),
    );
  }
}
