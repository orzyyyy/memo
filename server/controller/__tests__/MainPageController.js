import MainPageController from '../MainPageController';

describe('MainPageController', () => {
  it('get snapshots', () => {
    expect(MainPageController).toMatchSnapshot();
  });
});
