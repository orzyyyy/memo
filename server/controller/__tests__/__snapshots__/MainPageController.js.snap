// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageController get snapshots 1`] = `
Router {
  "methods": Array [
    "HEAD",
    "OPTIONS",
    "GET",
    "PUT",
    "PATCH",
    "POST",
    "DELETE",
  ],
  "opts": Object {
    "prefix": "",
  },
  "params": Object {},
  "stack": Array [
    Layer {
      "methods": Array [
        "HEAD",
        "GET",
      ],
      "name": null,
      "opts": Object {
        "end": true,
        "ignoreCaptures": undefined,
        "name": null,
        "prefix": "",
        "sensitive": false,
        "strict": false,
      },
      "paramNames": Array [],
      "path": "/",
      "regexp": /\\^\\(\\?:\\\\/\\(\\?=\\$\\)\\)\\?\\$/i,
      "stack": Array [
        [Function],
      ],
    },
    Layer {
      "methods": Array [
        "HEAD",
        "GET",
      ],
      "name": null,
      "opts": Object {
        "end": true,
        "ignoreCaptures": undefined,
        "name": null,
        "prefix": "",
        "sensitive": false,
        "strict": false,
      },
      "paramNames": Array [],
      "path": "/test",
      "regexp": /\\^\\\\/test\\(\\?:\\\\/\\(\\?=\\$\\)\\)\\?\\$/i,
      "stack": Array [
        [Function],
      ],
    },
  ],
}
`;
